src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/github/GithubExtension.kt

```diff
@@ -0,0 +1,89 @@
+package org.quiltmc.community.modes.quilt.extensions.github
+
+import com.expediagroup.graphql.client.ktor.GraphQLKtorClient
+import com.kotlindiscord.kord.extensions.checks.hasRole
+import com.kotlindiscord.kord.extensions.commands.Arguments
+import com.kotlindiscord.kord.extensions.commands.application.slash.publicSubCommand
+import com.kotlindiscord.kord.extensions.commands.converters.impl.int
+import com.kotlindiscord.kord.extensions.commands.converters.impl.string
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.publicSlashCommand
+import com.kotlindiscord.kord.extensions.types.respond
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import org.quiltmc.community.GITHUB_TOKEN
+import org.quiltmc.community.TOOLCHAIN_GUILD
+import org.quiltmc.community.TOOLCHAIN_MODERATOR_ROLE
+import org.quiltmc.community.github.generated.DeleteIssue
+import org.quiltmc.community.github.generated.FindIssueId
+import java.net.URL
+
+class GithubExtension : Extension() {
+    private val client = GraphQLKtorClient(URL("https://api.github.com/graphql"),
+        HttpClient(engineFactory = CIO, block = {
+            defaultRequest {
+                header("Authorization", "bearer $GITHUB_TOKEN")
+            }
+        })
+    )
+    override val name = "github"
+    override suspend fun setup() {
+        publicSlashCommand {
```

Comment:
As you may have seen in eg `UtilityExtension`, we're registering these to all guilds in a specific way. for example:

```kotlin
GUILDS.forEach { guildId ->
    ephemeralMessageCommand {
        name = "Raw JSON"

        guild(guildId)

        check { hasBaseModeratorRole() }

        action {
            // ...
        }
    }
}
```

This makes it available to everyone with the base `Moderators` role, which includes trainee moderators. If that's what you meant, great, but if you think it should be manager-locked, you can use `check { hasPermission(Permission.Administrator) }` instead. We're not using `allowRole` since it's essentially useless right now, with Discord planning on changing how slash command permissions work.

Additionally, please note how we space things out with newlines - it's important that everything is readable!

This command should probably also be ephemeral. Any reason why you picked a public command here?

Reply:
Yes, the scope of the command is correct. I'm using RFC 7 as guidance here--moderators aren't explicitly limited to one platform, and the RFC seems to imply that means that moderators have powers everywhere that RFC 7 applies. If that's not what you intended, I'd recommend amending the RFC.

I think I'll convert the command type to an ephemeral, but see if there's a way I can make a log of the action somewhere? As it happens right now I don't think there's a way to tell who actually requested the issue being deleted.

Reply:
It is what I intended - that said, there's a difference between full moderators and trainee moderators, so I wasn't 100% sure whether you intended on **trainee** moderators being able to run this command.

The command should be available on both servers; less room for human error. The toolchain server now has a `#github-logs` channel, available with the ID `906285091481849876` (so default to that, but allow configuration). There's an extension function in the filtering extension that you can base off of.